### PR TITLE
Data for files

### PR DESCRIPTION
**Pull request by ptbrowne**
If you put a JSON file right next to an HTML file with the same name, the HTML components will get the data as context.

### CODE REVIEW

**Comment by webpro** [comment]
Thanks for the PR, @ptbrowne! I like the idea, but I'm going to request a few changes in the code before we can merge it.

File: server/index.js [hunk 1]
```diff
@@ -111,14 +112,13 @@ module.exports = function start(options) {
                 var server = app.listen(app.get('port'), function() {
                     var host = server.address().address,
                         port = server.address().port;
-                    console.log('Styleguide server started at http://%s:%s', host, port)
```
Comment (webpro) [inline server/index.js:114]: I like to leave this line in, as it provides feedback things are running (also, it's not related to the rest of this PR).

File: server/util.js [hunk 2]
```diff
@@ -71,6 +75,16 @@ function getCapitalizedString(s) {
     }).join(' ')
 }
 
+function getDataForFile (filepath) {
+    var jsonPath = filepath.replace(/.html$/, '.json');
```
Comment (webpro) [inline server/util.js]: Please camelCase `filePath` (just ilke `jsonPath`).

File: server/util.js [hunk 3]
```diff
@@ -23,13 +26,14 @@ function getComponentsInfo(options) {
         matches = tplName.match(re);
         componentType = matches ? matches[1] : options.rootName;
         componentName = matches ? matches[2] : tplName.replace(/\.html/, '');
+        var dataForFile = getDataForFile(path.resolve(options.componentDir, tplName));
```
Comment (webpro) [inline server/util.js]: This is within a loop, so the `var` should be declared outside (just like e.g. `componentName`).

File: server/index.js (hunk 1 above)
Comment (ptbrowne) [inline server/index.js:114]: Yes sorry, I don't know why I removed it ><

**Comment by ptbrowne** [comment]
@webpro I followed your recommendations ;-) Thanks for reviewing my PR.

**Comment by webpro** [comment]
Thanks for changes @ptbrowne.

**Pull request merged**